homeassistant/components/switch/ecoal_boiler.py

```diff
@@ -0,0 +1,108 @@
+"""
+Allows to configuration ecoal (esterownik.pl) pumps as switches.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/switch.ecoal_boiler/
+"""
+import logging
+from typing import Optional
+
+import voluptuous as vol
+
+from homeassistant.components.switch import PLATFORM_SCHEMA
+from homeassistant.components.ecoal_boiler import DATA_ECOAL_BOILER
+from homeassistant.helpers.entity import ToggleEntity
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+# Available ids
+PUMP_IDNAMES = (
+    "central_heating_pump",
+    "domestic_hot_water_pump",
+    "central_heating_pump2",
+)
+
+ENABLED_SCHEMA = {}
+for pump_id in PUMP_IDNAMES:
+    ENABLED_SCHEMA[vol.Optional(pump_id)] = cv.string
+CONF_ENABLE = "enable"
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {vol.Optional(CONF_ENABLE): ENABLED_SCHEMA}
+)
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up switches based on ecoal interface."""
+    ecoal_contr = hass.data[DATA_ECOAL_BOILER]
+    config_enable = config.get(CONF_ENABLE, {})
+    # _LOGGER.debug("config_enable: %r", config_enable)
+    switches = []
+    for pump_id in PUMP_IDNAMES:     # pylint: disable=W0621
+        name = config_enable.get(pump_id)
+        if name:
+            switches.append(EcoalSwitch(ecoal_contr, name, pump_id))
+    add_devices(switches, True)
+
+
+class EcoalSwitch(ToggleEntity):
```

Comment:
Inherit from SwitchDevice in the switch component.

Reply:
Done.